fsthttp/transport.go

```diff
@@ -0,0 +1,97 @@
+package fsthttp
+
+import (
+	"net/http"
+	"strings"
+)
+
+// Transport is an http.RoundTripper implementation for backend requests
+// on Compute@Edge.
+//
+// Compute@Edge requests must be made to a pre-configured named backend.
+// Transport provides a mechanism for mapping hostnames to backend
+// names.  A default catchall backend is set when the Transport is
+// created, but additional host-to-backend mappings can be added with
+// the AddHostBackend method.
+//
+// Transport is provided primarily to adapt existing code which uses
+// http.Client instances to work on Compute@Edge.  Using an http.Client
+// pulls in substantially more code, resulting in slower compile times
+// and larger binaries.  For this reason, we recommend new code use the
+// fsthttp.Request type and its Send() method directly whenever
+// possible.
+type Transport struct {
+	defaultBackend string
+	backends       map[string]string
+
+	// Request is an optional callback invoked before the request is
+	// sent to the backend.  It allows callers to set
+	// fsthttp.Request-specific fields, such as cache control options.
+	Request func(req *Request) error
+}
+
+// NewTransport creates a new Transport instance with the given default
+// backend.  Any request made to a host not explicitly mapped to a
+// backend using the AddHostBackend method will be sent to the default
+// backend.
+func NewTransport(backend string) *Transport {
+	return &Transport{
+		defaultBackend: backend,
+		backends:       make(map[string]string),
+	}
+}
+
+// AddHostBackend adds a new host-to-backend mapping.  Multiple hosts
+// may be mapped to the same backend.
+func (t *Transport) AddHostBackend(host, backend string) {
+	t.backends[strings.ToLower(host)] = backend
```

Comment:
Do we want to protect `t.backends` with a mutex?

Reply:
I don't have a strong opinion on this.  It seems unlikely to me that users would add mappings concurrently, so my vote is to wait until it becomes a problem in practice.